File: 02_ProgrammingFundamentals/_06_ObjectsAndClasses/_01_Lecture/_04_Songs.java
package _02_ProgrammingFundamentals._06_ObjectsAndClasses._01_Lecture;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class _04_Songs {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int n = Integer.parseInt(scan.nextLine());

        // Add variable
        List<Song> songsList = new ArrayList<>();

        // Save the songs
        for (int i = 0; i < n; i++) {
            String[] tokens = scan.nextLine().split("_");

            String type = tokens[0];
            String name = tokens[1];
            String time = tokens[2];

            Song song = new Song(type, name, time);
            songsList.add(song);
        }

        String consoleType = scan.nextLine();

        // Print result
        if (consoleType.equals("all")) {
            for (Song element : songsList) {
                System.out.println(element);
            }
        } else {
            for (Song element : songsList) {
                if (element.getType().equals(consoleType)) {
                    System.out.println(element);
                }
            }
        }

        scan.close();
    }
}

// Create practice Song class
class Song {
    private String type;
    private String name;
    private String time;

    public Song(String type, String name, String time) {
        this.type = type;
        this.name = name;
        this.time = time;
    }

    public String getType() {
        return this.type;
    }

    @Override
    public String toString() {
        return this.name;
    }
}

File: 24_C#Fundamentals/_06_ObjectsAndClasses/_01_Lecture/_04_Songs/_04_Songs.cs
using System;

public class _04_Songs
{
    public static void Main(string[] args)
    {
        // Read input value
        int n = int.Parse(Console.ReadLine());

        // Add variable
        List<Song> songsList = new List<Song>();

        // Save the songs
        for (int i = 0; i < n; i++)
        {
            string[] tokens = Console.ReadLine().Split('_');

            string type = tokens[0];
            string name = tokens[1];
            string time = tokens[2];

            Song song = new Song(type, name, time);
            songsList.Add(song);
        }

        string consoleType = Console.ReadLine();

        // Print result
        if (consoleType.Equals("all"))
        {
            foreach (Song element in songsList)
            {
                Console.WriteLine(element);
            }
        }
        else
        {
            foreach (Song element in songsList)
            {
                if (element.getType().Equals(consoleType))
                {
                    Console.WriteLine(element);
                }
            }
        }
    }
}

// Create practice Song class
class Song
{
    private string Type { get; }
    private string Name { get; }
    private string Time { get; }

    public Song(string type, string name, string time)
    {
        this.Type = type;
        this.Name = name;
        this.Time = time;
    }

    public String getType()
    {
        return this.Type;
    }

    public override String ToString()
    {
        return this.Name;
    }
}
